spec/workers/articles/publish_worker_spec.rb

```diff
@@ -1,104 +1,120 @@
 require "rails_helper"
 
 RSpec.describe Articles::PublishWorker, type: :worker do
+  let(:worker) { subject }
+  let!(:article) { create(:article, published: true) }
+
   include_examples "#enqueues_on_correct_queue", "medium_priority"
 
-  describe "#perform" do
-    let(:worker) { subject }
-    let!(:article) { create(:article, published: true) }
+  it "calls Slack::Messengers::ArticlePublished to send slack notifications" do
+    allow(Slack::Messengers::ArticlePublished).to receive(:call)
+    worker.perform
+    expect(Slack::Messengers::ArticlePublished).to have_received(:call).with(article: article)
+  end
 
-    it "calls Slack::Messengers::ArticlePublished to send slack notifications" do
-      allow(Slack::Messengers::ArticlePublished).to receive(:call)
-      worker.perform
-      expect(Slack::Messengers::ArticlePublished).to have_received(:call).with(article: article)
-    end
+  it "sends notifications to mentioned users and followers" do
+    allow(Notification).to receive(:send_to_mentioned_users_and_followers)
+    worker.perform
+    expect(Notification).to have_received(:send_to_mentioned_users_and_followers).with(article)
+  end
 
-    it "sends notifications to mentioned users and followers" do
-      allow(Notification).to receive(:send_to_mentioned_users_and_followers)
+  it "schedules Notifications::NotifiableActionWorker" do
+    args = [article.id, "Article", "Published"]
+    sidekiq_assert_enqueued_with(job: Notifications::NotifiableActionWorker, args: args) do
       worker.perform
-      expect(Notification).to have_received(:send_to_mentioned_users_and_followers).with(article)
     end
+  end
+
+  it "doesn't send notifications for an old article" do
+    old_article = create(:article, :past, published: true, past_published_at: 1.year.ago)
+    allow(Notification).to receive(:send_to_mentioned_users_and_followers)
+    worker.perform
+    expect(Notification).not_to have_received(:send_to_mentioned_users_and_followers).with(old_article)
+  end
+
+  it "doesn't send notifications for a scheduled article" do
+    scheduled_article = create(:article, published: true, published_at: 1.day.from_now)
+    allow(Notification).to receive(:send_to_mentioned_users_and_followers)
+    worker.perform
+    expect(Notification).not_to have_received(:send_to_mentioned_users_and_followers).with(scheduled_article)
+  end
```

Comment:
I would add a spec "doesn't send moderation notification for a scheduled article"